syntax = "proto3";

package producer;

// Base event wrapper (matches Zombi's Event schema)
message Event {
  bytes payload = 1;
  int64 timestamp_ms = 2;
  string idempotency_key = 3;
  map<string, string> headers = 4;
}

// Domain events - serialized into Event.payload

message UserEvent {
  string event_type = 1;  // "signup", "login", "logout", "profile_update"
  string user_id = 2;
  string email = 3;
  string ip_address = 4;
  string user_agent = 5;
  map<string, string> metadata = 6;
}

message OrderEvent {
  string event_type = 1;  // "created", "paid", "shipped", "delivered", "cancelled"
  string order_id = 2;
  string user_id = 3;
  repeated OrderItem items = 4;
  int64 total_cents = 5;
  string currency = 6;
  string status = 7;
}

message OrderItem {
  string product_id = 1;
  string name = 2;
  int32 quantity = 3;
  int64 price_cents = 4;
}

message ClickstreamEvent {
  string session_id = 1;
  string user_id = 2;
  string page_url = 3;
  string referrer = 4;
  string action = 5;  // "pageview", "click", "scroll", "form_submit"
  string element_id = 6;
  int64 viewport_width = 7;
  int64 viewport_height = 8;
}

message MetricEvent {
  string metric_name = 1;
  double value = 2;
  string unit = 3;
  map<string, string> tags = 4;
}
